lib/matplotlib/axes/_axes.py

```diff
@@ -2394,27 +2395,110 @@ def stem(self, *args, **kwargs):
+        to *y*, and places a marker there.
+
         Call signatures::
 
-          stem(y, linefmt='b-', markerfmt='bo', basefmt='r-')
-          stem(x, y, linefmt='b-', markerfmt='bo', basefmt='r-')
+          stem([x,] y, **kwargs)
+          stem([x,] y, linefmt, **kwargs)
+          stem([x,] y, linefmt, markerfmt, **kwargs)
+          stem([x,] y, linefmt, markerfmt, basefmt, **kwargs)
+
+        The x-positions are optional. The formats may be provided either as
+        positional or as keyword-arguments.
+
+        Parameters
+        ----------
+        x : array-like, optional
+            The x-positions of the stems. Default: (0, 1, ..., len(y) - 1).
+
+        y : array-like
+            The y-values of the stem heads.
+
+        linefmt : str, optional
+            A string defining the properties of the vertical lines. Usually,
+            this will be a color or a color and a linestyle:
+
+            =========  =============
+            Character  Line Style
+            =========  =============
+            ``'-'``    solid line
+            ``'--'``   dashed line
+            ``'-.'``   dash-dot line
+            ``':'``    dotted line
+            =========  =============
+
+            Default: 'C0-', i.e. solid line with the first color of the color
+            cycle.
+
+            Note: While it is technically possible to specify valid formats
+            other than color or color and linestyle (e.g. 'rx' or '-.'), this
+            is beyond the intention of the method and will most likely not
+            result in a reasonable reasonable plot.
+
+        markerfmt : str, optional
+            A string defining the properties of the markers at the stem heads.
+            Default: 'C0o', i.e. filled circles with the first color of the
+            color cycle.
 
-        A stem plot plots vertical lines (using *linefmt*) at each *x*
-        location from the baseline to *y*, and places a marker there
-        using *markerfmt*.  A horizontal line at 0 is plotted using
-        *basefmt*.
+        basefmt : str, optional
+            A format string defining the properties of the baseline.
 
-        If no *x* values are provided, the default is (0, 1, ..., len(y) - 1)
+            Default: 'C3-' ('C2-' in classic mode).
+
+        bottom : float, optional, default: 0
+            The y-position of the baseline.
+
+        label : str, optional, default: None
+            The label to use for the stems in legends.
+
+
+        Other Parameters
+        ----------------
+        **kwargs
+            No other parameters are supported. They are currently ignored
+            silently for backward compatibility. This behavior is deprecated.
+            Future versions will not accept any other parameters and will
+            raise a TypeError instead.
+
+
+        Returns
+        -------
+        a :class:`~matplotlib.container.StemContainer`
+
+        The stemcontainer may be treated like a tuple
```

Comment:
should probably be indented under "a StemContainer"?  rendered version looks wrong otherwise.